src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -481,7 +513,10 @@ public virtual void Shutdown()
 
         private void OnHostHealthCheckTimer(object state)
         {
-            if (State == ScriptHostState.Running && !IsHostHealthy())
+            bool isHealthy = IsHostHealthy();
+            _healthCheckWindow.AddEvent(isHealthy);
+
+            if (!isHealthy && State == ScriptHostState.Running)
             {
```

Comment:
Can we write out a log message indicating why we're initiating a restart? It looks like when it starts back up it'll log the threshold that has been exceeded, but I can see it being a little confusing when you see a restart in our logs and it's not clear why that's happened.

Reply:
Good idea, done